sgl-kernel/csrc/cpu/topk.cpp

```diff
@@ -252,29 +252,33 @@ void topk_softmax_kernel_impl(
   });
 }
 
-template <typename scalar_t, int SIZE>
+template <typename param_t, int SIZE>
 inline void
-apply_bias(float* __restrict__ scores2, const float* __restrict__ scores, const scalar_t* __restrict__ bias) {
-  using bVec = at::vec::Vectorized<scalar_t>;
+apply_bias(float* __restrict__ scores2, const float* __restrict__ scores, const param_t* __restrict__ bias) {
   using fVec = at::vec::Vectorized<float>;
-  for (int d = 0; d < SIZE; d += bVec::size()) {
-    bVec bias_vec = bVec::loadu(bias + d);
-    fVec bias0, bias1;
-    std::tie(bias0, bias1) = at::vec::convert_to_float(bias_vec);
-
-    fVec x0 = fVec::loadu(scores + d) + bias0;
-    fVec x1 = fVec::loadu(scores + d + fVec::size()) + bias1;
+  using bVec = at::vec::Vectorized<param_t>;
+  auto vec_size = fVec::size() * 2 == bVec::size() ? bVec::size() : fVec::size() * 2;
```

Comment:
isn't this always 32 on avx512 for `vec_size`? just use:

```
template <typename scalar_t, typename param_t, int SIZE>
inline void apply_bias(...) {
  using bVec = att::vec::Vectorized<scalar_t>;
}
```
will do all the work.

Reply:
Yes, got the point, and refiend.